src/System.Memory/tests/ReadOnlyMemory/Strings.cs

```diff
@@ -0,0 +1,175 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Buffers;
+using System.Collections.Generic;
+
+namespace System.MemoryTests
+{
+    public static partial class ReadOnlyMemoryTests
+    {
+        public static IEnumerable<object[]> StringInputs()
+        {
+            yield return new object[] { "" };
+            yield return new object[] { "a" };
+            yield return new object[] { "a\0bcdefghijklmnopqrstuvwxyz" };
+        }
+
+        [Theory]
+        [MemberData(nameof(StringInputs))]
+        public static void AsReadOnlyMemory_ToArray_Roundtrips(string input)
+        {
+            ReadOnlyMemory<char> m = input.AsReadOnlyMemory();
+            Assert.Equal(input, new string(m.ToArray()));
+        }
+
+        [Theory]
+        [MemberData(nameof(StringInputs))]
+        public static void AsReadOnlyMemory_Span_Roundtrips(string input)
+        {
+            ReadOnlyMemory<char> m = input.AsReadOnlyMemory();
+            ReadOnlySpan<char> s = m.Span;
+            Assert.Equal(input, new string(s.ToArray()));
+        }
+
+        [Theory]
+        [InlineData("", 0, 0)]
+        [InlineData("0123456789", 0, 0)]
+        [InlineData("0123456789", 10, 0)]
+        [InlineData("0123456789", 0, 10)]
+        [InlineData("0123456789", 1, 9)]
+        [InlineData("0123456789", 2, 8)]
+        [InlineData("0123456789", 9, 1)]
+        [InlineData("0123456789", 1, 8)]
+        [InlineData("0123456789", 5, 3)]
+        public static void AsReadOnlyMemory_Slice_MatchesSubstring(string input, int offset, int count)
+        {
+            ReadOnlyMemory<char> m = input.AsReadOnlyMemory();
+            Assert.Equal(input.Substring(offset, count), new string(m.Slice(offset, count).ToArray()));
+            Assert.Equal(input.Substring(offset, count), new string(m.Slice(offset, count).Span.ToArray()));
+            if (offset + count == input.Length)
```

Comment:
Why is this condition needed?

Reply:
For the inputs where input.Substring(start) == input.Substring(start, count), it's testing both.

Reply:
Shouldn't the Assert succeed even if `offset + count != input.Length`?

Reply:
> Shouldn't the Assert succeed even if offset + count != input.Length?

Hmm.  Yes, actually.  I'll remove the guard.